core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -499,6 +499,10 @@ public void process(@Nullable MappingMetaData mappingMd, String concreteIndex) {
             }
         }
 
+        if ("".equals(id)) {
```

Comment:
Sounds good to me, I think it falls into the larger input validation category anyway